router/worker.go

```diff
@@ -1198,3 +1207,78 @@ func (w *worker) countTransformedJobStatuses(transformType string, transformedJo
 		}).Count(count)
 	}
 }
+
+// recordTransformerOutgoingRequestMetrics records both transformer_outgoing_request_latency and transformer_outgoing_request_count metrics
+// for router deliveries to match transformer's metric structure
+func (w *worker) recordTransformerOutgoingRequestMetrics(
+	postParams integrations.PostParametersT,
+	destinationJob types.DestinationJobT,
+	resp *routerutils.SendPostResponse,
+	duration time.Duration,
+) {
+	// Only emit metrics if EndpointPath is present
+	if postParams.EndpointPath == "" {
+		return
+	}
+
+	labels := deliveryMetricLabel{
+		DestType:      w.rt.destType,
+		EndpointPath:  postParams.EndpointPath,
+		StatusCode:    strconv.Itoa(resp.StatusCode),
+		RequestMethod: postParams.RequestMethod,
+		Module:        "router",
+		WorkspaceID:   destinationJob.Destination.WorkspaceID,
+		DestinationID: destinationJob.Destination.ID,
+	}
+
+	// Get or create cached stats objects
+	latencyStat := w.getOrCreateLatencyStat(labels)
+	countStat := w.getOrCreateCountStat(labels)
+
+	// Record metrics using cached stats
+	latencyStat.SendTiming(duration)
+	countStat.Increment()
+}
+
+// deliveryMetricLabel represents a unique key for caching stats based on labels
+type deliveryMetricLabel struct {
+	DestType      string `mapstructure:"destType"`
+	EndpointPath  string `mapstructure:"endpointPath"`
+	StatusCode    string `mapstructure:"statusCode"`
+	RequestMethod string `mapstructure:"requestMethod"`
+	Module        string `mapstructure:"module"`
+	WorkspaceID   string `mapstructure:"workspaceId"`
+	DestinationID string `mapstructure:"destinationId"`
+}
+
+// ToStatsTags converts metricCacheKey to stats.Tags using mapstructure
+func (w *worker) convertDeliveryMetricLabelToStatsTags(labels deliveryMetricLabel) stats.Tags {
+	var tags map[string]string
+	err := mapstructure.Decode(labels, &tags)
```

Comment:
This uses reflection, i.e. it is unnecessarily slow. We can simply do the mapping manually
```suggestion
func (dml *deliveryMetricLabel) ToStatsTags() stats.Tags {
```

Reply:
Done